### PR TITLE
package npm binaries, and prove the concept by packaging elm-graphql

### PR DESCRIPTION
**Pull request by BrianHicks**
I'm opening this to ask a question: would npm stuff belong here so that it can be packaged separately, rather than combined in the monolith/monorepo?

### CODE REVIEW

**Comment by avh4** [comment]
The only thing I see added here is elm-graphql -- am I missing something?  -- can we contribute that upstream to nix-pkgs or to elm-graphql itself so that our repos don't need custom workarounds?

**Comment by BrianHicks** [comment]
I just chose that as one that doesn't have any actual binary stuff to download.

I've just been wondering about how to manage npm dependencies and development tools in the same project, and came out with "oh duh, the dev tools can be separate derivations."

Unless I've misunderstood, this is our staging ground for contributing stuff upstream. Is that incorrect?

File: default.nix [hunk 1]
```diff
@@ -15,4 +15,5 @@
   overlays = import ./overlays; # nixpkgs overlays
 
   adr-tools = pkgs.callPackage ./pkgs/adr-tools { };
+  elm-graphql = pkgs.callPackage ./pkgs/elm-graphql { };
```
Comment (ento) [inline default.nix:18]: As a staging space before pushing expressions up to the nixpkgs repo, it'll make it easier to make upstream PRs if we use [the same structure as nixpkgs's `nodePackages`](https://github.com/NixOS/nixpkgs/blob/49fb21232ff08a2f591237a97ebbbd7beda8d9c2/pkgs/top-level/all-packages.nix#L3955-L3957). According to [the docs](https://github.com/NixOS/nixpkgs/blob/49fb21232ff08a2f591237a97ebbbd7beda8d9c2/doc/languages-frameworks/node.section.md), `pkgs.nodePackages` contain end user software packages, which matches what we're trying to extract with this PR.

**Comment by ento** [comment]
> would npm stuff belong here so that it can be packaged separately, rather than combined in the monolith/monorepo?

Until we designate Nix as the one and only dev environment manager, I think it's best to use package.json as the single source of truth for managing the versions of dev tools. Or: we could make the Nix side fail if versions don't match between package.json and what will be installed by Nix.

That aside, I think it makes sense to not include binary stuff in the node-packages.nix of library dependencies needed to build the frontend.

>  this is our staging ground for contributing stuff upstream

Yes, that is the assumed ideal. There's no process that enforces that we contribute stuff upstream though. I'm still not sure if having this repo is very bad, very good, or a necessary evil

**Comment by BrianHicks** [comment]
ah, got it. It sounds like this may not have been a feasible idea after all. I certainly learned more about `node2nix` doing it though, so hooray for that!

**Pull request closed without merging**